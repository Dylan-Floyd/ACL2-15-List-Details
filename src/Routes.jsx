import React from 'react'
import { Route, Switch } from 'react-router-dom'
import DetailView from './views/DetailView/DetailView.jsx'
import Home from './views/Home.jsx'
import Layout from './views/Layout.jsx'
import ListView from './views/ListView/ListView.jsx'

export default function Routes() {
  return (
    <Layout>
      <Switch>
        <Route path='/characters/:id'>
          <DetailView />
        </Route>
        <Route path='/characters'>
          <ListView />
        </Route>
        <Route path='/'>
          <Home />
        </Route>
      </Switch>
    </Layout>
  )
}
