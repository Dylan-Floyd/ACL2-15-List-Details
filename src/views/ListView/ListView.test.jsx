import { render, screen } from '@testing-library/react'
import { rest } from 'msw'
import { setupServer } from 'msw/node'
import { MemoryRouter } from 'react-router-dom'
import Routes from '../../Routes.jsx'

const server = setupServer(
  rest.get('https://rickandmortyapi.com/api/character', (req, res, ctx) => {
    return res(
      ctx.json({
        results: [
          {
            id: 1,
            name: 'bob',
            origin: 'earth',
            status: 'hi erich',
            image: 'asdf.jpg'
          },
          {
            id: 2,
            name: 'bob2',
            origin: 'earth',
            status: 'hi erich',
            image: 'asdf.jpg'
          }
        ]
      })
    )
  })
)

beforeAll(() => {
  server.listen()
})

afterAll(() => {
  server.close()
})

it('should fetch and render a list of characters', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/characters']}>
      <Routes />
    </MemoryRouter>
  )

  screen.getByText('Loading')
  await screen.findByText('bob')
  await screen.findByText('bob2')

  expect(container).toMatchSnapshot()
})