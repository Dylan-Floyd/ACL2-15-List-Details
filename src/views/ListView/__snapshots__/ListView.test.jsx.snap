// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should fetch and render a list of characters 1`] = `
<div>
  <header>
    <h1>
      Rick and Morty Characters
    </h1>
    <h4>
      <a
        href="/"
      >
        Home
      </a>
      <span>
         / 
        <a
          href="/characters"
        >
          characters
        </a>
      </span>
    </h4>
  </header>
  <div
    class="charList"
  >
    <a
      class="clickable"
      href="/characters/1"
    >
      <div
        class="charCard"
      >
        <h3
          class="name"
        >
          bob
        </h3>
        <div
          class="origin"
        >
          Origin: 
        </div>
        <div>
          Status: 
          hi erich
        </div>
        <img
          alt="bob"
          src="asdf.jpg"
        />
      </div>
    </a>
    <a
      class="clickable"
      href="/characters/2"
    >
      <div
        class="charCard"
      >
        <h3
          class="name"
        >
          bob2
        </h3>
        <div
          class="origin"
        >
          Origin: 
        </div>
        <div>
          Status: 
          hi erich
        </div>
        <img
          alt="bob2"
          src="asdf.jpg"
        />
      </div>
    </a>
  </div>
</div>
`;
